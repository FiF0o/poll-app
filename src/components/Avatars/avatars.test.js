import React from 'react'
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import List from 'material-ui/List/List';
import ListItem from 'material-ui/List/ListItem';
import Avatar from 'material-ui/Avatar';

import { UserAvatar } from './'

describe('<UserAvatar /> component', () => {
    let component
    let stub
    // const noop = (e) => e

    beforeEach(() => {
        stub = {
            user: {
                photoURL: '',
                email: '',
                uid: '',
                displayName: ''
            }
        }
        component = shallow(
            <UserAvatar {...stub} />
        )
    })

    it('should render a <UserAvatar />', () => {
    //    expect(component.find(List).length).toBe(1)
       expect(component.containsMatchingElement(List)).toBe(true)
    })

    it('should render correctly', () => {
        const uStub = {
            user: {
                photoURL: 'abc.jpg',
                email: 'a@mail.com',
                uid: 'qwe',
                displayName: 'aaa'
            }
        }
        const tree = renderer.create(
            <MuiThemeProvider>
                <UserAvatar {...uStub} />
            </MuiThemeProvider>
          ).toJSON();
          expect(tree).toMatchSnapshot();
    })

})
