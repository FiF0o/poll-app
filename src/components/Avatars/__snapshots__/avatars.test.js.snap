// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserAvatar /> component should render correctly 1`] = `
<div
  style={
    Object {
      "muiPrepared": true,
      "padding": "8px 0px 8px 0px",
    }
  }
>
  <div>
    <div
      onTouchEnd={[Function]}
      open={null}
      style={
        Object {
          "MozTransition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "WebkitTransition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "backgroundColor": null,
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "block",
          "fontSize": 16,
          "lineHeight": "16px",
          "marginLeft": 0,
          "muiPrepared": true,
          "paddingBottom": 20,
          "paddingLeft": 72,
          "paddingRight": 16,
          "paddingTop": 20,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    >
      aaa
       - 
      a@mail.com
      <img
        className={undefined}
        size={40}
        src="abc.jpg"
        style={
          Object {
            "MozUserSelect": "none",
            "WebkitAlignItems": "center",
            "WebkitBoxAlign": "center",
            "WebkitBoxPack": "center",
            "WebkitJustifyContent": "center",
            "WebkitUserSelect": "none",
            "alignItems": "center",
            "backgroundColor": "rgb(188, 188, 188)",
            "borderRadius": "50%",
            "color": "#ffffff",
            "display": Array [
              "-webkit-inline-box",
              "-moz-inline-box",
              "-ms-inline-flexbox",
              "-webkit-inline-flex",
              "inline-flex",
            ],
            "fontSize": 20,
            "height": 40,
            "justifyContent": "center",
            "left": 16,
            "msFlexAlign": "center",
            "msFlexPack": "center",
            "msUserSelect": "none",
            "muiPrepared": true,
            "position": "absolute",
            "top": 8,
            "userSelect": "none",
            "width": 40,
          }
        }
      />
    </div>
  </div>
</div>
`;
